### ISSUE TITLE
React devtools 4.0.4 (8/18/2019) Expected to find root ID

### ISSUE THREAD

**Issue by techird** [facebook/react#16500]
**Do you want to request a *feature* or report a *bug*?**

a bug

**What is the current behavior?**

The new dev tools is deprecating react 0.14 in a breaking way.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**

Reference to [react-devtools#1388](https://github.com/facebook/react-devtools/issues/1388), the stack:

![stack](https://user-images.githubusercontent.com/4995935/63243613-abcc9700-c28d-11e9-8350-9cb4b8dac2d4.png)

It happend when you try to unmount element in React 14.

**What is the expected behavior?**

Just give a dprecated warning instead of breaking the runtime.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

React 0.14.7

**Comment by MonstraG** [comment]
Can confirm, React 0.14.0.

**Comment by zasuh** [comment]
Can also confirm, React 0.14.0

**Comment by kunukn** [comment]
Breaking change.

> 
> Which versions of React are supported?
> react-dom
> 
> 0-14.x: Not supported
> 15.x: Supported (except for the new component filters feature)
> 16.x: Supported
> react-native
> 
> 0-0.61: Not supported
> 0.62: Will be supported (when 0.62 is released)

https://reactjs.org/blog/2019/08/15/new-react-devtools.html#which-versions-of-react-are-supported

**Comment by bvaughn** [comment]
As @kunukn mentioned, the new DevTools does not support React v14 and older. [This was mentioned in the blog post](https://reactjs.org/blog/2019/08/15/new-react-devtools.html#which-versions-of-react-are-supported) along with [instructions on how to install the previous version of the DevTools](https://reactjs.org/blog/2019/08/15/new-react-devtools.html#how-do-i-get-the-old-version-back).

I apologize for the inconvenience here. We're a small team and we really only have one or two people working on DevTools most of the time. Since version 15 has been out for over 3 years, we made a pragmatic decision to only invest in supporting v15 and newer.

Thanks for understanding.

**Comment by bvaughn** [comment]
Note related issue #16462 - We do plan to _improve_ the v14 experience.

**Comment by prasanthLalapeta** [comment]
Deprecated warning will be good instead of code breaks :)

**Comment by pinkdoremi** [comment]
If you won't support v14, you can just show me an Error just like 'not support any more'. But now, it breaks my projects. 
I use react in many projects vary from v14 to 16.7. The only way for me to debug the v14 project is to disable the React devtools temporarily.

**Comment by bvaughn** [comment]
As mentioned two comments above yours, @pinkdoremi

> we do plan to _improve_ the v14 experience by showing a notification (rather than just throwing). That work is covered in #16462.

**Comment by bibblebabl** [comment]
Same. `"react": "^0.14.6"`

**Comment by gaearon** [comment]
I'll lock this because "+1"s are not helpful and don't add any new information. In case you've missed the reply: https://github.com/facebook/react/issues/16500#issuecomment-523087105

TLDR:

* 0.14 is not supported in the new DevTools
* If you need it, you can build old DevTools from source and keep using them as long as you like: https://reactjs.org/blog/2019/08/15/new-react-devtools.html#how-do-i-get-the-old-version-back

We'll add a better error message in the future.

Hope this helps!
